libs/service-portal/access-control/delegations/src/hooks/useDomains.ts

```diff
@@ -80,25 +80,50 @@ export const useDomains = (includeDefaultOption = true) => {
     sessionStore.setItem('domain', name)
 
     const query = new URLSearchParams(location.search)
+    const currentDomain = query.get('domain')
 
-    if (query.get('domain')) {
+    if (currentDomain && currentDomain !== displayNameQueryParam) {
       query.set('domain', name)
       history.push(`${location.pathname}?${query.toString()}`)
     }
   }
 
+  const updateDomainByName = (name: string) => {
+    const option = getOptionByName(name)
+
+    // Priority
+    // 1. Option is found by name
+    // 2. Option is not found by name, try to find ISLAND_DOMAIN
+    // 3. ISLAND_DOMAIN option is not found, select the first option in the list
+    if (option) {
+      updateDomain(option)
+    } else {
+      const islandDomainOption = getOptionByName(ISLAND_DOMAIN)
+
+      // Default to ISLAND_DOMAIN if the domain is not found
+      if (islandDomainOption) {
+        updateDomain(islandDomainOption)
+      } else {
+        // Default to the first option in the list
+        updateDomain(options[0])
+      }
+    }
+  }
+
   useEffect(() => {
     if (data?.authDomains) {
       const sessionDomainName = sessionStore.getItem('domain')
 
-      if (sessionDomainName) {
-        setDomainName(sessionDomainName)
+      // Priority
+      // 1. Query string
+      // 2. Session storage
+      // 3. Default domain
+      if (displayNameQueryParam) {
+        updateDomainByName(displayNameQueryParam ?? ISLAND_DOMAIN)
```

Comment:
~Already checking `displayNameQueryParam`, so it should never fallback to `?? ISLAND_DOMAIN`.